docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,194 @@
+
+Proper contextual support for original data is not only a concern for retrieval but also for generation. Even powerful LLMs are heavily sensitive to document structure. Lack of proper segmentation or unusual structure will result in poor tokenization (typically, merging a document separator with a token) and a higher rate of omissions and hallucinations. With tables, a hacky (but verbose) fix I’ve found is to recall the row name/column name couple before each entry.
+
+Finally, retrieval metrics don't necessarily fit the data. Embedding models rely on a fuzzy concept of “similarity” that is likely to fail the last mile. They are trained on specific sets of documents and questions (typically bases like MS-Marco). This will not necessarily extend well to specialized resources, and, more fundamentally, similarity may not align with user expectations.
+
+### 2.4 Settling for hybrid
+As RAG systems naturally evolve with longer documents, enriched with additional context and more and more uninitiated users, simply transposing their usual search habits, the reality sets in: embedding models in RAG systems can barely compete with classic information retrieval approaches like BM25 and have to be supplemented by it (embedding models have to be supplemented by classic info retrieval approaches).
+
+What this means in practice:
+* Hybrid indexation is emerging as the new standard: on my latest internal benchmarks, combining a keyword-based search with embedding similarity hit the highest retrieval rate (nearly 90% of resources found among the top ten results). The easiest implementation I’ve found so far is lancedb’s, but the jury's still out - there's a lot of activity in this space right now.
+* Keyword-based search is still a strong alternative. The infrastructure is already there and at a very large scale, embeddification can quickly become prohibitive. [Keyword-based search can still be a strong alternative, especially when dealing with longer texts or more expansive context information.]
+* Evaluation has to be built from the ground up, based on the constraints of your existing infrastructure (especially with regard to data quality/indexation) and the ways users interact with it.
+
+Overall, there is a lot of alpha left in recovering information retrieval methods and indicators in an LLM context, especially as many key features can now be accurately generated/extracted at scale. Jo Kristian Bergum from Vespa has just introduced a [very convincing method of data expansion](https://blog.vespa.ai/improving-retrieval-with-llm-as-a-judge/), by grounding an LLM-as-a-Judge on a small relevant dataset, that can be transformed into a large one. Basically, generative AI doesn't replace the classic approaches of retrieval evaluation; it can completely reshape their logistics. Intensive data work that would have been available only to large scale organizations is suddenly scalable with far fewer resources.
+
```

Comment:
The context suggests that your example will illustrate how to use older retrieval methods and indicators at scale. Is this an example of using an older info retrieval method / indicator?

Reply:
I think the article makes this point but could be better presented: "Jo Kristian Bergum from Vespa has very convincingly shown how classic metrics of information retrieval (precision at k, recall) and overall evaluation design can be effectively repurposed with emerging practices from the fields, such as LLM-as-a-Judge"